pyctbgui/services/Acquisition.py

```diff
@@ -312,38 +323,108 @@ def setDBITPipeline(self):
+        self.view.checkBoxFileWriteRaw.setChecked(self.det.fwrite)
+        self.view.checkBoxFileWriteRaw.stateChanged.connect(self.setFileWrite)
 
     def setFileWrite(self):
-        self.det.fwrite = self.view.checkBoxFileWrite.isChecked()
+        self.det.fwrite = self.view.checkBoxFileWriteRaw.isChecked()
         self.getFileWrite()
 
+    def setFileWriteNumpy(self):
+        """
+        slot for saving the data in numpy (.npy) format
+        """
+        self.writeNumpy = not self.writeNumpy
+
     def getFileName(self):
+        """
+        set the lineEditFilePath input widget to the filename value from the detector
+        """
+
         self.view.lineEditFileName.editingFinished.disconnect()
-        self.view.lineEditFileName.setText(self.det.fname)
+        fileName = self.det.fname
+        self.view.lineEditFileName.setText(fileName)
+        self.outputFileNamePrefix = fileName
         self.view.lineEditFileName.editingFinished.connect(self.setFileName)
 
     def setFileName(self):
+        """
+        slot for setting the filename from the widget to the detector
+        """
         self.det.fname = self.view.lineEditFileName.text()
         self.getFileName()
 
     def getFilePath(self):
+        """
+        set the lineEditFilePath input widget to the path value from the detector
+        """
         self.view.lineEditFilePath.editingFinished.disconnect()
-        self.view.lineEditFilePath.setText(str(self.det.fpath))
+        path = self.det.fpath
+        self.view.lineEditFilePath.setText(str(path))
         self.view.lineEditFilePath.editingFinished.connect(self.setFilePath)
+        self.outputDir = path
 
     def setFilePath(self):
+        """
+        slot to set the directory of the output for the detector
+        """
         self.det.fpath = Path(self.view.lineEditFilePath.text())
         self.getFilePath()
 
+    def saveNumpyFile(self, data: np.ndarray | dict[str, np.ndarray], jsonHeader):
+        """
+        save the acquisition data (waveform or image) in the specified path
+        save waveform in multiple .npy files
+        save image as npy file format
+        @note: frame number can be up to 100,000 so the data arrays cannot be fully loaded to memory
+        """
+        if not self.writeNumpy:
+            return
+        if self.outputDir == Path('/'):
+            self.outputDir = Path('./')
+        if self.outputFileNamePrefix == '':
+            self.outputFileNamePrefix = 'run'
+
+        for device in data:
+            if device not in self.numpyFileManagers:
+                tmpPath = self.outputDir / f'{self.outputFileNamePrefix}_{device}_{jsonHeader["fileIndex"]}.npy'
+                self.numpyFileManagers[device] = NumpyFileManager(tmpPath, 'w', data[device].shape, data[device].dtype)
+            self.numpyFileManagers[device].writeOneFrame(data[device])
+        self.closeOpenedNumpyFiles(jsonHeader)
+
+    def closeOpenedNumpyFiles(self, jsonHeader):
```

Comment:
Could you think of a better name for this function? Maybe in line with closeNumpyFilesIfDone? or something like that? Reading the code at the call site it looks like the function closes the files and you have to look in the implementation to see that the files are only closed under certain conditions.

Reply:
instead of renaming the function I moved the condition to `saveNumpyFile()` for better readability